tests/fields/test_permissions.py

```diff
@@ -50,3 +53,31 @@ def user(self) -> str:
     assert definition.fields[0].python_name == "user"
     assert definition.fields[0].graphql_name is None
     assert definition.fields[0].permission_classes == [IsAuthenticated]
+
+
+@pytest.mark.asyncio
+async def test_permission_classes_async():
+    class IsAuthenticated(BasePermission):
+        message = "User is not authenticated"
+
+        @sync_to_async
+        def has_permission(
+            self, source: Any, info: strawberry.Info, **kwargs: Any
+        ) -> bool:
+            return True
+
+    @sync_to_async
+    def resolver() -> str:
+        return "patrick"
+
+    @strawberry.type
+    class Query:
+        user: str = strawberry.field(
+            resolver=resolver, permission_classes=[IsAuthenticated]
+        )
+
+    schema = strawberry.Schema(Query)
+
+    result = await schema.execute("query { user }")
+
```

Comment:
The bug's root cause is actually in strawberry.resolver, so we also need a test in `test_resolver.py`, checking if `StrawberryResolver.is_async` works as expected. This test is fine as a regression test for your specific issue 😊

Reply:
I just added a test case for `StrawberryResolver.is_async`. Please look at the updated PR.

Reply:
thanks, looks good now!